New docs with monorepo support

0b5vr opened the pull request:
This PR introduces the new docs page with the monorepo support.

![image](https://github.com/user-attachments/assets/5fba014b-4c36-4f26-9ec5-2f4e6cfcfa7b)

Existing docs for each package still exist and can be generated by running the `docs-legacy` command. These docs redirect to the new URL.
It consumes a long time to generate. We might want to remove it in the future.

Also, existing `/docs/` in the root level is moved to `/guides/`.

See details for the each commit.

0b5vr committed:
docs: move root `/docs/` into `/guides/`

to place typedoc on the root level

0b5vr committed:
docs: on README of `packages/three-vrm`, the link to CONTRIBUTING.md and LICENSE was broken. fixed this

0b5vr committed:
docs: New unified docs

existing docs commands still exist as `docs-legacy` in order to preserve the exising URLs
I will try to redirect them in the next commit

Set `name` for each packages' `typedoc.json` to make their URL not being like `_pixiv_three-vrm`

`/guides/README.md` is removed since it appears as "README" on the docs menu and it doesn't make sense

Also updated the inspect.yml to build and publish both old and new docs
I have 20% confidence this ci will work on the first try

0b5vr committed:
docs: redirect from legacy docs url to new docs url

used typedoc plugin. Referred the existing redirect plugin
See: https://github.com/Gerrit0/typedoc-plugin-redirect/blob/main/src/plugin.ts

0b5vr committed:
chore: `yarn all` now runs `docs-legacy`

also `yarn clean` now cleans the root `docs/`

yue4u commented on typedoc-redirect-legacy-docs-plugin.mjs:
```diff
@@ -0,0 +1,78 @@
+  const matchItems = path.match(/packages\/(.+)\/docs\/(classes|interfaces|functions|types|variables)\/(.+)/);
+  if (matchItems != null) {
+    const [_, packageName, kind, item] = matchItems;
+    return `../../../docs/${kind}/${packageName}.${item}`;
```
The relative path `../../../` does not work for nested docs like `/packages/three-vrm/docs/variables/VRMExpressionMaterialColorType-1.html`. Maybe using `/docs/` is better?

0b5vr replied:
Oops that's true. will change it to simply use `/docs/...`.

0b5vr replied:
no that wouldn't work because the document root will be `https://pixiv.github.io/three-vrm/docs/` on GitHub Pages. I would try setting the document root using environment variable.

0b5vr replied:
make it use an environment variable `DOCS_ROOT`.

https://github.com/pixiv/three-vrm/pull/1588/commits/8f9f9ae309409bc2ce1badeef1df90cf9b0dc606

yue4u reviewed (commented):
IMO the new doc is much better and keeping redirects is good!

0b5vr committed:
fix: Fix redirect of docs-legacy

the previous redirect using the relative path does not work when the page is nested
Use `DOCS_ROOT` environment variable to specify the correct absolute path

https://github.com/pixiv/three-vrm/pull/1588#discussion_r1982796832

yue4u reviewed (approved):
LGTM

0b5vr committed:
ci: gh-pages job should wait for docs-legacy job to finish
